### PR TITLE
Trivial comments and util tweaks.

### PR DESCRIPTION
**Pull request by cconvey**
## Description ##
(Brief description on what this PR is about)

### Essentials ###
- [ ] Passed code style checking (`make lint`)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: src/ngraph/ngraph_utils.cc [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+namespace ngraph_bridge {
+
+std::ostream& operator<<(std::ostream& os, const ngraph::Shape& s) {
+  os << "[";
+  for (size_t d : s) {
+    os << " " << d;
+  }
+  os << "]";
+  return os;
+}
```
Comment (mbrookhart) [inline src/ngraph/ngraph_utils.cc]: I've done this a lot while debugging, the utilty is nice. Can we templatize it to support nnvm::Shape as well?

Comment (cconvey) [reply src/ngraph/ngraph_utils.cc]: I've added an overload to handle `nnvm::TShape`.  I'm hesitant to templatize this in a way that picks up both `nnvm::TShape` and `ngraph::Shape`, because I fear such a template would bind to uses that we never intended.

File: src/ngraph/ngraph_utils.cc [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+namespace ngraph_bridge {
+
+std::ostream& operator<<(std::ostream& os, const ngraph::Shape& s) {
+  return container_to_debug_stream(os, s);
+}
+
+std::ostream& operator<<(std::ostream& os, const nnvm::TShape& s) {
+  container_to_debug_stream(os, s);
+  return os;
```
Comment (mbrookhart) [inline src/ngraph/ngraph_utils.cc]: nitpick:
`return container_to_debug_stream(os, s);`?

Comment (mbrookhart) [reply src/ngraph/ngraph_utils.cc]: Doesn't match the other implementation.

Comment (cconvey) [reply src/ngraph/ngraph_utils.cc]: Oh sweet mother of pearl... I'm going to need to restart CI because of that mistake.
